sopel/irc/backends.py

```diff
@@ -428,7 +508,25 @@ async def _run_forever(self) -> None:
         # nothing to read anymore
         LOGGER.debug('Shutting down writer.')
         self._writer.close()
-        await self._writer.wait_closed()
+        try:
+            await self._writer.wait_closed()
+
+        # task was cancelled, i.e. another exception is responsible for that
+        except asyncio.CancelledError:
+            LOGGER.debug('Writer task was cancelled.')
+
+        # connection reset happened before on the read task
+        except ConnectionResetError as err:
+            LOGGER.debug('Connection reset while closing: %s', err)
+
+        # generic (connection) error requires a specific exception log
+        except ConnectionError as err:
+            LOGGER.error('Connection error while closing: %s', err)
+            self.log_exception()
+        except Exception:
```

Comment:
```suggestion

        except Exception:
```

There's no good reason for this, except that it looks weird to me when every other `except` block gets its own pre-gap.